File: training-data/julia/edit_distance.jl
function edit_distance(a::String, b::String)
    m, n = length(a), length(b)
    table = zeros(Int, m + 1, n + 1)
    for i in 0:m
        table[i + 1, 1] = i
    end
    for j in 0:n
        table[1, j + 1] = j
    end
    for i in 1:m
        for j in 1:n
            cost = a[i] == b[j] ? 0 : 1
            table[i + 1, j + 1] = min(
                table[i, j + 1] + 1,
                table[i + 1, j] + 1,
                table[i, j] + cost
            )
        end
    end
    return table[m + 1, n + 1]
end

println(edit_distance("kitten", "sitting"))

File: training-data/ocaml/edit_distance.ml
let edit_distance a b =
  let m = String.length a in
  let n = String.length b in
  let table = Array.make_matrix (m + 1) (n + 1) 0 in
  for i = 0 to m do table.(i).(0) <- i done;
  for j = 0 to n do table.(0).(j) <- j done;
  for i = 1 to m do
    for j = 1 to n do
      let cost = if a.[i - 1] = b.[j - 1] then 0 else 1 in
      table.(i).(j) <- min (min (table.(i - 1).(j) + 1) (table.(i).(j - 1) + 1)) (table.(i - 1).(j - 1) + cost)
    done
  done;
  table.(m).(n)

let () = print_int (edit_distance "kitten" "sitting"); print_newline ()
